File: cpp/getKthFromEnd.cpp
class Solution {
public:
    ListNode* getKthFromEnd(ListNode* head, int k) {
        ListNode* newHead = head;
        int i = 0;
        while(i < k){
            head = head->next;
            i++;
        }
        while(head){
            head = head->next;
            newHead = newHead->next;
        }
        return newHead;

    }
};

File: getKthFromEnd.py
class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        newHead = head
        for _ in range(k):
            head = head.next
        
        while head:
            head = head.next
            newHead = newHead.next
        return newHead
            